File: 필수문항/149_인간-컴퓨터상호작용.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder ans = new StringBuilder();
        StringTokenizer st;
        String S = br.readLine();
        int[][] alphabet = new int[26][S.length()];
        alphabet[(int) S.charAt(0) - 97][0] = 1;
        for (int i = 1; i < S.length(); i++) {
            alphabet[(int) S.charAt(i) - 97][i] = 1;
            for (int[] alpha : alphabet) {
                alpha[i] += alpha[i - 1];
            }
        }

        int q = Integer.parseInt(br.readLine());
        while (q-- > 0) {
            st = new StringTokenizer(br.readLine());
            int c = (int) st.nextToken().charAt(0) - 97;
            int start = Integer.parseInt(st.nextToken());
            int end = Integer.parseInt(st.nextToken());
            if (start == 0) {
                ans.append(alphabet[c][end]).append('\n');
            } else {
                ans.append(alphabet[c][end] - alphabet[c][start - 1]).append('\n');
            }
        }
        System.out.println(ans);
    }
}

File: 필수문항/149_인간-컴퓨터상호작용.py
import sys
from collections import defaultdict

input = sys.stdin.readline

S = input().rstrip()
alphabet = defaultdict(lambda: [0] * len(S))

alphabet[S[0]][0] = 1

for i in range(1, len(S)):
    alphabet[S[i]][i] = 1
    for alpha in alphabet.values():
        alpha[i] += alpha[i - 1]

q = int(input())
for _ in range(q):
    c, start, end = input().split()
    start = int(start)
    end = int(end)
    if start == 0:
        print(alphabet[c][end])
    else:
        print(alphabet[c][end] - alphabet[c][start - 1])
